Initialize authentication service with JWT security

E-Line21 opened the pull request:
- **Starter Auto-Configuration**: Adds a Spring Boot starter for JWT authentication with customizable properties.
- **Security Integration**: Implements a SecurityConfigurer and registers security filter chains for access and refresh tokens.
- **Token Services**: Provides composite services for access and refresh tokens (generation, parsing, rotation, invalidation).
- **Authentication Components**:
   - Filters for JSON login, access token, and refresh token processing.
   - Providers for validating access and refresh tokens with automatic rotation.
   - Success and failure handlers that emit standardized JSON or cookie responses.
- **Error Handling**: Centralized error response generation with context-aware HTTP status mapping.
- **Persistence & Claims Extraction**: In-memory and JDBC-style token repositories, claim extractors (including UUID and authorities).
- **Testing**: Comprehensive integration and unit tests covering all authentication modes, filters, handlers, and services.

Commit:
Init jwt-serivice-library module

Commit:
Add JWT service interfaces for token handling

- Introduce JwtGeneratorService interface with token generation capabilities
  - Supports adding custom claims via Map parameter
  - Utilizes Spring Security UserDetails for principal information

- Create JwtParserService interface for token validation and claims extraction
  - Provides default implementations for common claim extraction
  - Includes token validation with AuthenticationException handling

- Add composite JwtService interface combining generation and parsing
  - Unifies JWT operations under single service contract
  - Inherits method contracts from parent interfaces

Commit:
Implement RSA JWT generator service with comprehensive tests

- RsaJwtGeneratorService implementing Jwt Generator Service interface
  - UseRS 256 algorithm with private key signing
  - Support configurable expiration duration and issuer claim
  - Add optional extra claims validation with builder flag
  - Implement expiration date calculation based on Duration

- Add validation for:
  - Extra claims allowance (with IllegalArgumentException)

- Include test suite

Commit:
Add RSA JWT parser service with comprehensive validation

- Implement RsaJwtParserService for JWT token validation and claims extraction
  - Use RS256 algorithm with public key signature verification
  - Validate issuer claim, token expiration, and subject presence
  - Translate JWT exceptions to Spring Security authentication exceptions
  - Support claims extraction through default interface methods

- Implement exception hierarchy:
  │-- CredentialsExpiredException (expired tokens)
  │-- InsufficientAuthenticationException (missing subject)
  └-- BadCredentialsException (general validation failures)

- Include test suite

Commit:
Add composite JWT service implementation

- Introduce CompositeJwtService combining generator and parser
  - Implements Jwt Service interface through component delegation
  - Maintains strict separation between generation and parsing concerns

- Implement pure delegation architecture:
  ├── Generation → JwtGeneratorService
  ├── Validation → JwtParserService
  └── Claims extraction → JwtParserService

- Include test suite

Commit:
Init spring-boot-starter-jwt-authentication

Commit:
Add access token service interfaces

- Introduce AccessTokenGeneratorService interface with token generation with UserDetails and custom
  claims

- Create AccessTokenParserService interface for:
  - Core token validation with AuthenticationException hierarchy
  - Claims extraction

- Add composite AccessTokenService combining:
  - Generation capabilities from AccessTokenGeneratorService
  - Parsing/validation from AccessTokenParserService

Commit:
Add refresh token service interfaces with fingerprint binding

- Introduce RefreshTokenGeneratorService interface for fingerprint-bound refresh token generation

- Create RefreshTokenParserService interface with:
  - Token validation against client fingerprint
  -  UUID extraction for token tracking

- Implement RefreshTokenService combining:
  - Token generation capabilities
  - Validation and claims extraction
  - Token invalidation method

Commit:
Add identifiable UserDetails extension

- Introduce Identifiable User Details interface extending Spring Security's UserDetails
  - Adds getId() method for user identifier access

Commit:
Refactor AccessTokenParserService

- Created a functional ClaimExtractor interface to encapsulate the extraction of a claim from a
  token

- Refactored method `extractClaim(token, claimExtractor)` user ClaimExtractor

Commit:
Add token qualifier and access token generator implementation

- Introduce @Token qualifier annotation for DI token type discrimination
  - TokenType enum with ACCESS/REFRESH classification
  - Targets all injectable components (fields, methods, params, types)

- Implement JwtAccessTokenGeneratorService features
  - Automatic security context claim injection
    - User ID via IdentifiableUserDetails
    - Authority list from GrantedAuthority

- Include test suite

Commit:
Add access token parser service implementation

- Implement JwtAccessTokenParserService as access token processing facade

- Include test suite

Commit:
Add composite access token service implementation

- Implement CompositeAccessTokenService coordinating generator and parser

- Include test suite

Commit:
Add refresh token persistence components

- Implement RefreshTokenEntity model
  - UUID-based primary key
  - Temporal metadata for token rotation

- Create RefreshTokenRepository interface with core operations:
  - save: Generates unique ID and persists entity
  - findById: Optional-based retrieval
  - deleteById: Idempotent deletion
  - existById: Existence check

- Add InMemoryRefreshTokenRepository implementation based on ConcurrentHashMap

- Include test suit

Commit:
Implement refresh token generator with fingerprint binding

- Add JwtRefreshTokenGeneratorService with features
  - Persists hashed client fingerprints
  - Integrates with refresh token repository

- Include test suite

Commit:
Add configurable claim extractor and UUID claim extractor implementation

- Implement claim extraction hierarchy:
  - ConfigurableClaimExtractor: Runtime claim name configuration
  - AbstractClaimExtractor: Template method pattern base
  - AbstractConfigurableClaimExtractor: Mutable claim name support
  - UuidClaimExtractor: UUID-specific parsing logic

- UUID extraction features:
  - Handles both UUID objects and strings
  - Graceful null returns for invalid formats

- Include test suite

Commit:
Implement refresh token parser with fingerprint validation

- Add JwtRefreshTokenParserService with features
  - Full refresh token lifecycle validation
  - Fingerprint hash matching via PasswordEncoder
  - Token existence checks in repository

- Include test suite

Commit:
Add authentication token Authentication implementations

- Implement AbstractJwtTokenAuthenticationToken base class:
  - Enforces authentication state transition rules
  - Manages JWT credentials storage
  - Provides pre-authentication principal constant

- Add concrete implementations:
  - AccessAuthenticationToken: Standard access token handling
  - RefreshAuthenticationToken: Fingerprint-bound refresh tokens
  - Factory methods for safe instance creation

- Include test suite

Commit:
Implement composite refresh token service

- Add CompositeRefreshTokenService with full lifecycle management:
  - Token generation via RefreshTokenGeneratorService
  - Validation and parsing via RefreshTokenParserService
  - Persistence operations via RefreshTokenRepository

- Include test suite

Commit:
Add JWT authentication login components with JSON request support

- Introduce `LoginRequest` DTO to encapsulate login credentials (username, password, fingerprint)

- Implement `TokenResponseUsernamePasswordAuthenticationFilter`:
  - Custom Spring Security filter to extract credentials from JSON request body
  - Replaces form-based auth with JSON payload parsing via `RequestMapper`

- Add `RequestMapper` interface and `ObjectMapperRequestMapper` implementation:
  - Generic utility for deserializing HTTP request bodies using Jackson

- Include test suite

Commit:
Add standardized error handling for authentication failures

- Implement ErrorResponse DTO with validation builder

- Create AbstractErrorResponseAuthenticationFailureHandler:
  - Centralized error response generation
  - Configurable status/message mapping

- Add ErrorResponseUsernamePasswordAuthenticationFailureHandler:
  - Maps 10+ Spring Security exceptions to proper status codes
  - Prevents sensitive error leakage in responsesk

Commit:
Add secure refresh token cookie management

- Implement RefreshTokenCookieWriter interface and SimpleRefreshTokenCookieWriter implementation

- Include test suite

Commit:
Add token response handling for successful logins

- Implement TokenResponseAuthenticationSuccessHandler:
  - Generates access/refresh tokens on login
  - Supports JSON and cookie response formats

- Modify DTOs for authentication workflow:
  - LoginRequest: Credentials + fingerprint
  - FingerprintRequest: Base fingerprint container
  - TokenResponse: Standardized token container

- Include test suite

Commit:
Add token-aware UserDetails implementations

- Implement AccessTokenUserDetails for JWT authentication:
  - Passwordless UserDetails for token contexts
  - Null-safe authority handling

- Add AccessTokenIdentifiableUserDetails:
  - Extends base with persistent user ID
  - Implements IdentifiableUserDetails interface
  - Generic type support for ID flexibility

Commit:
Add JWT access token authentication filter

- Implement AccessTokenAuthenticationProcessingFilter

- Include test suite

Commit:
Add access token authentication provider

- Implement AccessTokenAuthenticationProvider features:
  - Token validation and principal resolution
  - Configurable claim extraction for authorities/user ID
  - Support for identifiable user details

- Include test suite

Commit:
Add refresh token cookie reader implementation

- Implement RefreshTokenCookieReader interface

- Add SimpleRefreshTokenCookieReader

- Include test suite

Commit:
Add refresh token authentication filter

- Implement TokenResponseRefreshTokenAuthenticationFilter

- Include test suite

Commit:
Add authorities claim extractor implementation

- Implement AuthoritiesClaimExtractor for token authority claim processing

- Include test suite

Commit:
Add token-specific authentication failure handler

- Implement ErrorResponseTokenAuthenticationFailureHandler with:
  - Context-aware error messages using token type
  - Custom HTTP status mapping for token errors

- Include test suite

Commit:
Add refresh token authentication provider

- Implement RefreshTokenAuthenticationProvider with:
  - JWT refresh token validation
  - Automatic token rotation (invalidate after use)
  - UserDetails integration for security context
  - Exception translation for security events

- Include test suite

Commit:
Add refresh token invalidation logout handler

- Implement RefreshTokenInvalidationLogoutHandler with:
  - Dual token source handling (cookie/body)
  - Fingerprint-validated token revocation
  - Configurable cookie cleanup

- Include test suite

Commit:
Add JWT authentication configuration properties

- Implement JwtProperties with nested configuration structure:
  - Global settings: mode, endpoints, issuer
  - Token configurations: access/refresh expiration, headers, cookies
  - Cryptographic key settings: algorithm, key resources

- Key configuration options:
  - MODE: OFF/SERVER/CLIENT operation modes
  - Token expiration with DurationUnit support
  - Access token header/prefix customization
  - Refresh token cookie configuration

Commit:
Add security configurer functional interface

- Introduce SecurityConfigurer functional interface for custom security configuration via
  HttpSecurity

Commit:
Add auto-configuration classes for JWT authentication starter

- Implement SecurityConfig to configure security filter chain integration

- Support external configuration via application.yaml

Commit:
Add integration tests for JWT starter client, server and off modes

- Verify bean creation for JWT server, client and off mode configuration

Commit:
Init authentication-service

Commit:
Fix authentication starter

Problem 1: InvalidDefinitionException during deserialization
Cause:
- LoginRequest used a 3-parameter constructor and Jackson could not map JSON fields to
  constructor parameters
Fix:
- Added Lombok's @Jacksonized to complement @SuperBuilder

Problem 2: Missing auto-configuration registration
Cause:
- Spring Boot could not detect custom auto-configurations because
  org.springframework.boot.autoconfigure.AutoConfiguration.imports was missing
Fix:
- Added src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports

Problem 3: Inputstream closed exception
Cause:
- Servlet requests only allow single read of InputStream/Reader. This caused failures in
  authentication processors
Fix:
- Introduce CachedBodyHttpServletRequestWrapper to cache HTTP request body content in byte
  array for repeated access
- Implement CachingFilter with high priority (-106) to wrap requests early in chain

Commit:
Add security configuration and temporary auth setup

- Configure JWT authentication using custom SecurityConfigurer from starter
- Implement in-memory UserDetailsService stub with admin user credentials
  (temporary solution until User Microservice integration)

Commit:
Implement Redis key utility for refresh tokens

- Add RefreshTokenRedisKeyUtil interface and implementation
- Support configurable key prefix via properties

Commit:
Implement Redis-based refresh token repository

- Add RedisRefreshTokenRepository with CRUD operations
- Configure RedisTemplate for refresh tokens

rbrmnv commented on modules/jwt-service-library/src/main/java/com/technokratos/eateasy/jwtservice/JwtParserService.java:
```diff
@@ -0,0 +1,54 @@
+     * @return username contained in the token
+     * @throws IllegalArgumentException if token is malformed or invalid
+     */
+    default String extractUsername(String token) throws IllegalArgumentException {
```
Почему бы не использовать единый тип исключений? (Например, кастомное JwtException)

E-Line21 replied:
Потому что метод предполагает, что токен заведомо валидный, поэтому бросается не какая-то ошибка валидации, а именно `IllegalArgumentException`

rbrmnv commented on modules/jwt-service-library/src/main/java/com/technokratos/eateasy/jwtservice/JwtParserService.java:
token ничем не помечен – может ли он быть null?

E-Line21 replied:
Если программист считает целесообразным достать username из `null`, то может. NPE не будет, будет `IllegalArgumentException`

rbrmnv commented on modules/jwt-service-library/src/main/java/com/technokratos/eateasy/jwtservice/JwtService.java:
Почему разделил на два интерфейса (просто интересно)?

E-Line21 replied:
Потому что не всегда нужно иметь возможность и выдавать токен, и парсить его. Выдавать может только Authentication Service, а остальные могут только читать, поэтому если был бы один интерфейс, то пришлось бы бросать какой-нибудь `UnsupportedOperationException`

rbrmnv commented on modules/jwt-service-library/src/main/java/com/technokratos/eateasy/jwtservice/impl/RsaJwtGeneratorService.java:
```diff
@@ -0,0 +1,100 @@
+    public String generate(UserDetails userDetails, @NonNull Map<String, Object> extraClaims) {
+        if (!supportsExtraClaims && !Collections.isEmpty(extraClaims)) {
+            log.error("Token generation failed: extra claims are not supported");
+            throw new IllegalArgumentException("Extra claims are not supported");
```
Исключение бы кастомное сюда

rbrmnv commented on modules/jwt-service-library/src/main/java/com/technokratos/eateasy/jwtservice/impl/RsaJwtParserService.java:
```diff
@@ -0,0 +1,141 @@
+                throw new InsufficientAuthenticationException("Token subject is missing or blank");
+            }
+
+        } catch (ExpiredJwtException e) {
+            log.debug("Token validation failed: Token expired at {}",  e.getClaims().getExpiration());
+            throw new CredentialsExpiredException("Token has expired", e);
+
+        } catch (UnsupportedJwtException e) {
+            log.debug("Token validation failed: Unsupported JWT format - {}", e.getMessage());
+            throw new BadCredentialsException("Unsupported token format", e);
+
+        } catch (MalformedJwtException e) {
+            log.debug("Token validation failed: Malformed JWT - {}", e.getMessage());
+            throw new BadCredentialsException("Invalid token structure", e);
+
+        } catch (IncorrectClaimException e) {
+            log.debug("Token validation failed: Claim validation failed. Got '{}'='{}'",
+                    e.getClaimName(), e.getClaimValue());
+            throw new BadCredentialsException("Invalid token claims", e);
+
+        } catch (SignatureException e) {
+            log.debug("Token validation failed: Signature verification failed - {}", e.getMessage());
+            throw new BadCredentialsException("Invalid token signature", e);
+        } catch (JwtException e) {
+            log.debug("Token validation failed: General JWT exception - {}", e.getMessage());
+            throw new BadCredentialsException("Token validation failed", e);
+        }
```
Здесь все ошибки связаны с JWT, можно сделать JwtAuthenticationException и внутри него уже определять какая именно

E-Line21 replied:
Тут нельзя создать единое исключение, т.к. Spring Security предлагает свою иерархию. Самое лучше, что тут можно изобрести, это наследоваться от каждого исключения Spring Security и создавать новое, но как будто это бессмысленно.

rbrmnv replied:
Согласен

rbrmnv commented on modules/spring-boot-starter-jwt-authentication/build.gradle:
```diff
@@ -0,0 +1,46 @@
+dependencies {
+    implementation 'org.slf4j:slf4j-api'
+    implementation project(':modules:jwt-service-library')
+    compileOnly 'jakarta.servlet:jakarta.servlet-api'
```
уже есть же в testImplementation

E-Line21 replied:
У них разные области видимости. То, что есть в runtime, то попадает в test, но не наоборот. В runtime зависимость помечена как `compileOnly`, т.к. в рантайме ее предоставляет Tomcat, но в тестах ее никто не предоставляет, поэтому там нужно еще раз пометить ее как `testImplementation`, чтобы она была доступна в рантайме в момент исполнения тестов

rbrmnv commented on modules/spring-boot-starter-jwt-authentication/src/main/java/com/technokratos/eateasy/jwtauthenticationstarter/config/AuthenticationManagerConfig.java:
```diff
@@ -0,0 +1,45 @@
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnProperty(prefix = "jwt", name = "mode", havingValue = "client")
+    public AuthenticationManager clientAuthenticationManager() throws Exception {
+        AuthenticationManagerBuilder authenticationManagerBuilder =
```
Мб стоит добавить проверку на null

E-Line21 replied:
А что делать, если null? Я пробовал несколько способов, как создать и настроить `AuthenticationManager`, только этот заработал. В теории, тут может оказаться null, но непонятно, как реагировать? Исключение бросать? Оно и так будет выброшено

rbrmnv commented on modules/spring-boot-starter-jwt-authentication/src/main/java/com/technokratos/eateasy/jwtauthenticationstarter/config/AuthenticationManagerConfig.java:
```diff
@@ -0,0 +1,45 @@
+
+@Configuration
+@RequiredArgsConstructor
+@ConditionalOnExpression("'${jwt.mode:off}'.equals('off') == false")
```
Вот здесь можно вместо условия использовать параметр "havingValue"

E-Line21 replied:
А как с помощью `havingValue` указать "что угодно, кроме 'off'"? Просто я не придумал, как это сделать...

rbrmnv replied:
@ConditionalOnProperty(
    prefix = "jwt", 
    name = "mode", 
    havingValue = "off", 
    matchIfMissing = false,
    havingValueIsNot = true  
)

E-Line21 replied:
Круто, я даже не знал, что так можно

E-Line21 replied:
А нет. `havingValueIsNot` - такой настройки нет

rbrmnv replied:
Ладно тогда лучше твое оставить

rbrmnv commented on modules/spring-boot-starter-jwt-authentication/src/main/java/com/technokratos/eateasy/jwtauthenticationstarter/config/DefaultBeansConfig.java:
Можно вынести @ConditionalOnMissingBean на класс, тк аннотация есть на всех бинах

E-Line21 replied:
`@ConditionalOnMissingBean` на уровне класса будет отвечать за создание самого `ServerOnlyDefalultBeansConfig`. А так как это класс конфигурации и никто в здравом уме не будет создавать его вручную, то его никогда не будет, соответственно эта аннотация бесполезна. Кроме это, теперь все бины в этом конфиге будут создаваться **всегда**, а не только при их отсутствии, что может привести к конфликтам

rbrmnv commented on modules/spring-boot-starter-jwt-authentication/src/main/java/com/technokratos/eateasy/jwtauthenticationstarter/config/RequestCachingConfig.java:
```diff
@@ -0,0 +1,28 @@
+        FilterRegistrationBean<HttpServletRequestBodyCachingFilter> registrationBean = new FilterRegistrationBean<>();
+        registrationBean.setFilter(httpServletRequestBodyCachingFilter());
+        registrationBean.addUrlPatterns("/*");
+        registrationBean.setOrder(-106);
```
Дублирование, в аннотации уже есть порядок

rbrmnv commented on modules/spring-boot-starter-jwt-authentication/src/main/java/com/technokratos/eateasy/jwtauthenticationstarter/config/RequestCachingConfig.java:
```diff
@@ -0,0 +1,28 @@
+public class RequestCachingConfig {
+
+    @Bean
+    @Order(-106)
```
хардкод порядка

rbrmnv commented on modules/spring-boot-starter-jwt-authentication/src/main/java/com/technokratos/eateasy/jwtauthenticationstarter/dto/request/LoginRequest.java:
```diff
@@ -0,0 +1,40 @@
+        this.password = password;
+    }
+
+    public String login() {
+        return login;
+    }
+
+    public String password() {
+        return password;
+    }
```
а можно @Data и без этого))

E-Line21 replied:
Я хейтер `@Data`, `@Value`, `@UtilityClass` и т.д.)

rbrmnv commented on modules/spring-boot-starter-jwt-authentication/src/main/java/com/technokratos/eateasy/jwtauthenticationstarter/dto/request/RefreshRequest.java:
```diff
@@ -0,0 +1,26 @@
+
+    private final String refreshToken;
+
+    public RefreshRequest(String fingerprint, String refreshToken) {
+        super(fingerprint);
+        this.refreshToken = refreshToken;
+    }
+
+    public String refreshToken() {
+        return refreshToken;
+    }
```
Сюда тоже @Data и @AllArgsConstructor

E-Line21 replied:
Пробелема в том, что `@AllArgsConstructor` не работает с классами, у родителя которых конструктор с параметрами
<img width="515" alt="image" src="https://github.com/user-attachments/assets/128ffc9a-9d31-4a93-a9f1-81d848487a56" />

rbrmnv replied:
А с @RequaredArgsConst тоже самое?

E-Line21 replied:
Да

rbrmnv commented on modules/spring-boot-starter-jwt-authentication/src/main/java/com/technokratos/eateasy/jwtauthenticationstarter/properties/JwtProperties.java:
```diff
@@ -0,0 +1,149 @@
+    /**
+     * Authentication starter operation mode
+     */
+    @NotNull
+    private MODE mode = MODE.OFF;
+
+    /**
+     * Authentication endpoint paths
+     */
+    @NotEmpty
+    private String loginUrl = "/auth/login";
+
+    @NotEmpty
+    private String refreshUrl = "auth/refresh";
+
+    @NotEmpty
+    private String logoutUrl = "/auth/logout";
+
+    /**
+     * Issuer claim value for generated tokens
+     */
+    @NotEmpty
+    private String issuer = "app";
```
Как они могут быть пустыми, если уже заданы?

E-Line21 replied:
Если user обладает особым интеллектом, то он может переопределить их как пустая строка. Например, так:

```YAML
jwt:
  mode: server
  login-url: ""
  refresh-url: /api/v1/auth/refresh
  logout-url: /api/v1/auth/logout
```

rbrmnv commented on modules/spring-boot-starter-jwt-authentication/src/main/java/com/technokratos/eateasy/jwtauthenticationstarter/security/authentication/AbstractJwtTokenAuthenticationToken.java:
```diff
@@ -0,0 +1,89 @@
+    /**
+     * Principal identifier for pre-authenticated tokens
+     */
+    public static final String PRE_AUTH_PRINCIPAL = "JWT_PRE_AUTH_USER";
```
Вынести в проперти надо быы

E-Line21 replied:
Это просто заглушка, чтобы в моменте не было `null`. Пользователь не должен опираться на значение principal, чтобы проверить, что клиент аутентифицирован, для этого есть метод `Authentication#isAuthenticated()`, поэтому не нужна возможность конфигурировать это значение

rbrmnv commented on modules/spring-boot-starter-jwt-authentication/src/main/java/com/technokratos/eateasy/jwtauthenticationstarter/token/access/service/impl/JwtAccessTokenServiceConstants.java:
Вынести в .property файл

rbrmnv commented on modules/spring-boot-starter-jwt-authentication/src/main/java/com/technokratos/eateasy/jwtauthenticationstarter/token/refresh/repository/impl/InMemoryRefreshTokenRepository.java:
```diff
@@ -0,0 +1,76 @@
+@Slf4j
+public class InMemoryRefreshTokenRepository implements RefreshTokenRepository {
+
+    private final Map<UUID, RefreshTokenEntity> store = new ConcurrentHashMap<>();
+
```
А можно добавить максимальный размер для защиты от переполнения

E-Line21 replied:
Эта реализация не предполагает использования в production, поэтому нет нужды ее оптимизировать. Она нужна только для пробного запуска, где переполнение невозможно

> This implementation is not suitable for production use due to lack of persistence and scalability.

rbrmnv commented on modules/spring-boot-starter-jwt-authentication/src/main/java/com/technokratos/eateasy/jwtauthenticationstarter/token/refresh/service/impl/RefreshTokenServiceConstants.java:
```diff
@@ -0,0 +1,11 @@
+
+public final class RefreshTokenServiceConstants {
+
+    public static final String REFRESH_TOKEN_ID = "rti";
+    public static final String USER_ID = "uid";
```
Тоже в проперти

rbrmnv commented on modules/spring-boot-starter-jwt-authentication/src/main/java/com/technokratos/eateasy/jwtauthenticationstarter/utils/refreshtokencookiewriter/impl/SimpleRefreshTokenCookieWriter.java:
```diff
@@ -0,0 +1,74 @@
+                refreshTokenCookie.getMaxAge(), refreshTokenCookie.getPath());
+    }
+
+    private Cookie createCookie(String refreshToken) {
+        Cookie refreshTokenCookie = new Cookie(cookieName, refreshToken);
+        refreshTokenCookie.setMaxAge(calculateMaxAge(expiration));
+        refreshTokenCookie.setHttpOnly(true);
+        refreshTokenCookie.setPath(refreshUrl);
+        refreshTokenCookie.setAttribute("SameSite", "Strict");
+        refreshTokenCookie.setAttribute("Partitioned", Boolean.TRUE.toString());
+
+        return refreshTokenCookie;
```
можно билдер сделать

E-Line21 replied:
А как мне сделать builder на не мой класс? Это класс из Jakarta Servlet Api

Commit:
Added comprehensive integration tests for authentication service endpoints:
- /api/v1/auth/login
- /api/v1/auth/refresh
- /api/v1/auth/logout

Commit:
Integrate JWT keys generation into build lifecycle

- Add generateRsaKeys task to create RSA key pairs for:
  - access tokens (access_private.pem, access_public.pem)
  - refresh tokens (refresh_private.pem, refresh_public.pem)
- Make processTestResources depend on generateRsaKeys to ensure
  keys are generated before test execution
- Keys are placed in build/resources/test/security directory

Commit:
Extract RSA key generation into standalone plugin

- Moved RSA key pair generation logic from build.gradle to a dedicated Gradle plugin
- Created KeyGenerationTask class to handle key generation for access/refresh tokens

Commit:
Move hardcoded values to properties

- Extracted all string constants and default values to JwtProperties
- Fixed PR review comments

E-Line21 committed:
Merge branch 'develop' into feature/BRAEN-01/auth-service-init

E-Line21 committed:
Merge branch 'develop' into feature/BRAEN-01/auth-service-init

SmesiteJl commented on services/authentication-service/src/main/java/com/technokratos/eateasy/authenticationservice/util/RefreshTokenRedisKeyUtilImpl.java:
```diff
@@ -0,0 +1,26 @@
+@RequiredArgsConstructor
+public class RefreshTokenRedisKeyUtilImpl implements RefreshTokenRedisKeyUtil {
+
+    @Value("${custom.repository.refresh-token.key-prefix}")
```
тут не падает NPE? 
Вроде бы final поля должны быть инициализированы либо при объявлении, либо в конструкторе, а @Value инициализирует уже после создания объекта.

SmesiteJl commented on modules/spring-boot-starter-jwt-authentication/src/main/java/com/technokratos/eateasy/jwtauthenticationstarter/utils/refreshtokencookiereader/impl/SimpleRefreshTokenCookieReader.java:
```diff
@@ -0,0 +1,48 @@
+        Cookie[] cookies = request.getCookies();
+        if (Objects.isNull(cookies)) {
+            log.trace("No cookies found in the request.");
+            return null;
```
Лучше вернуть Optional<Cookie>, чтобы явно показать возможность отсутствия результата и избежать NullPointerException у вызывающего.

SmesiteJl commented on modules/spring-boot-starter-jwt-authentication/src/main/java/com/technokratos/eateasy/jwtauthenticationstarter/config/DefaultBeansConfig.java:
```diff
@@ -0,0 +1,66 @@
+    @Configuration
+    @RequiredArgsConstructor
+    @ConditionalOnProperty(prefix = "jwt", name = "mode", havingValue = "server")
+    public class ServerOnlyDefalultBeansConfig {
```
Опечатка 
<del>ServerOnlyDefalultBeansConfig</del> -> ServerOnlyDefaultBeansConfig
